arangod/MMFiles/MMFilesWalRecoverState.cpp

```diff
@@ -444,6 +444,35 @@ bool MMFilesWalRecoverState::InitialScanMarker(MMFilesMarker const* marker, void
 bool MMFilesWalRecoverState::ReplayMarker(MMFilesMarker const* marker,
                                           void* data, MMFilesDatafile* datafile) {
   MMFilesWalRecoverState* state = reinterpret_cast<MMFilesWalRecoverState*>(data);
+  auto visitHelpers = std::shared_ptr<MMFilesWalRecoverState>(
```

Comment:
there is no reason to use `shared_ptr` here, you can use TRI_DEFER(...) to ensure that function will be called eventually

Reply:
done